### PR TITLE
Add deduplication to grantImmediateUsageCredits

### PR DESCRIPTION
**Pull request by joeysabs**
## What Does this PR Do?

This PR implements deduplication in the `grantImmediateUsageCredits` function to prevent duplicate credit grants when the `addFeatureToSubscription` API is called multiple times with the same feature. The fix uses the stable `featureId` instead of the ephemeral `subscription_item_feature.id` for deduplication checks, ensuring credits are granted only once per feature per billing period.

## Key Changes

- Added deduplication check using JOIN query on stable `featureId`
- Handles both scenarios: with billing period and without billing period
- Added 4 comprehensive test cases covering edge cases and exploit prevention

## Test Coverage

All 29 tests pass, including the new deduplication tests that verify:
- Duplicate credits are not granted on repeated API calls
- Credits for different features are still granted independently
- Deduplication uses stable feature IDs correctly

---
## Summary by cubic
Prevents duplicate usage credits when the same feature is added multiple times. grantImmediateUsageCredits now deduplicates by stable featureId and scopes checks to the current billing period (or none).

- **Bug Fixes**
  - Join-based dedup using featureId, usageMeterId, and billingPeriodId (or isNull) on ManualAdjustment credits.
  - Skips issuing a ledger command when a matching credit exists, ensuring one credit per feature per billing period.
  - Adds tests for repeated/multi calls, different features, null and active billing periods, cross-period grants, and stable featureId dedup.

<sup>Written for commit f49bc2dedd511bc02b384a1ce9761bd8107fce27. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented duplicate usage credits from being generated when granting the same feature multiple times within a billing period, improving credit accuracy.

* **Tests**
  * Added extensive test suite validating deduplication logic across multiple scenarios including consecutive grants, different features, edge cases, and billing period handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/subscriptions/subscriptionItemFeatureHelpers.ts [hunk 1]
```diff
@@ -394,6 +397,49 @@ const grantImmediateUsageCredits = async (
...
+
+    // If we have a billing period, scope deduplication to that period
+    // If not, check for credits with null billingPeriodId
+    if (currentBillingPeriod) {
+      whereConditions.push(
+        eq(usageCredits.billingPeriodId, currentBillingPeriod.id)
+      )
+    }
+
+    const existingCredits = await transaction
```
[the remark this answers is not in the document]
Comment (agreea) [reply platform/flowglad-next/src/subscriptions/subscriptionItemFeatureHelpers.ts]: +1 for an `insert-on-conflict` pattern if it's easy to swing
